app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/ContactDetailsConfirmAddressController.scala

```diff
@@ -53,49 +53,29 @@ class ContactDetailsConfirmAddressController @Inject() (
 
   def displayPage(): Action[AnyContent] =
     (authenticate andThen journeyAction).async { implicit request =>
-      request.registration.incorpJourneyId match {
-        case Some(journeyId) =>
-          request.registration.organisationDetails.organisationType match {
-            case Some(UK_COMPANY) =>
-              executeAddressUpdate(journeyId,
-                                   _ =>
-                                     request.registration.organisationDetails.incorporationDetails match {
-                                       case Some(details) =>
-                                         updateBusinessAddress(details.companyAddress.toPptAddress)
-                                       case None =>
-                                         throw DownstreamServiceError(
-                                           "Failed to fetch incorporated details from cache",
-                                           RegistrationException(
-                                             "Failed to fetch incorporated details from cache"
-                                           )
-                                         )
-                                     }
-              )
-            case Some(SOLE_TRADER) =>
-              executeAddressUpdate(journeyId,
-                                   _ =>
-                                     updateBusinessAddress(
-                                       //TODO - temporary while we're working out where to get it from
-                                       Address(addressLine1 = "2 Scala Street",
-                                               addressLine2 = Some("Soho"),
-                                               townOrCity = "London",
-                                               postCode = "W1T 2HN"
-                                       )
-                                     )
+      request.registration.organisationDetails.organisationType match {
+        case Some(UK_COMPANY) =>
+          executeAddressUpdate(getIncorporationDetails())
+        case Some(SOLE_TRADER) =>
+          executeAddressUpdate(
+            updateBusinessAddress(
+              //TODO - temporary while we're working out where to get it from
+              Address(addressLine1 = "2 Scala Street",
+                      addressLine2 = Some("Soho"),
+                      townOrCity = "London",
+                      postCode = "W1T 2HN"
               )
-            case _ => throw new InternalServerException(s"Company type not supported")
-          }
-
-        case _ => Future(Redirect(routes.RegistrationController.displayPage()))
+            )
```

Comment:
I think we could have extracted this one into a private helper also, more for consistency than anything else.

Reply:
Also, perhaps we ought to perform the address update and the building of the page once we've obtained the appropriate address rather than duplicating that code/call in each case?

Perhaps this method ought to look more like:
1. Get the appropriate address (via private helper)
2. Update the address - specific call which makes it clear that this is our intention
3. Build the page